process/serialize.go

```diff
@@ -0,0 +1,237 @@
+	ser.Add("7", data.GetMap())
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Map) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", mapProcessNodeMapOutput(data.Outputs))
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_List) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", processNodeMapOutputs(data.Outputs))
+	return ser.HashSerialize()
+}
+
+type processNodeMapOutputs []*Process_Node_Map_Output
+
+// HashSerialize returns the hashserialized string of this type
+func (data processNodeMapOutputs) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	for i, value := range data {
+		ser.Add(strconv.Itoa(i), value)
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Reference) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.NodeKey)
+	ser.Add("2", data.Path)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Reference_Path) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.GetKey())
+	ser.AddInt("2", int(data.GetIndex()))
+	ser.Add("3", data.Path)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Filter) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("2", processNodeFilterConditions(data.Conditions))
+	return ser.HashSerialize()
+}
+
+type processNodeFilterConditions []Process_Node_Filter_Condition
+
+// HashSerialize returns the hashserialized string of this type
+func (data processNodeFilterConditions) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	for i, value := range data {
+		ser.Add(strconv.Itoa(i), value)
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data Process_Node_Filter_Condition) HashSerialize() string {
+	ser := hashserializer.New()
+	ser.AddString("1", data.Key)
+	ser.AddInt("2", int(data.Predicate))
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Edge) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.Src)
+	ser.AddString("2", data.Dst)
+	return ser.HashSerialize()
+}
+
+type processNodes []*Process_Node
+
+// HashSerialize returns the hashserialized string of this type
+func (data processNodes) HashSerialize() string {
```

Comment:
These function `processNodes` and `processEdges` could have a sort. Sorting the data in the case of a process will not change the behavior and the process (might slightly change the performance of it). As long as the process contains the same nodes and the same edges, whatever the order of the nodes or edges it will be the same process and the hash should be similar.

Reply:
why would you like to sort them? it's an array so the order is important.
but yes, from a process point of view, it doesn't change the behavior.
even, what would you like the array with? the value of the node key?
i think here it's up to the compiler to produce the same output.

Reply:
The idea would be to sort for example by `key`. The order of this array doesn't impact the actual structure of the graph.
It could definitely be done in the compiler too but might be interesting here as it's the final step but anyway this is not something related to this PR.

Reply:
could you create an issue to furthe discuss thus

Reply:
https://github.com/mesg-foundation/engine/issues/1707